integration_tests/run_pyspark_from_build.sh

```diff
@@ -117,9 +117,17 @@ else
     NUM_LOCAL_EXECS=${NUM_LOCAL_EXECS:-0}
     MB_PER_EXEC=${MB_PER_EXEC:-1024}
     CORES_PER_EXEC=${CORES_PER_EXEC:-1}
-
-    if ((NUM_LOCAL_EXECS > 0)); then
-      export PYSP_TEST_spark_master="local-cluster[$NUM_LOCAL_EXECS,$CORES_PER_EXEC,$MB_PER_EXEC]"
+    # Spark 3.1.1 includes https://github.com/apache/spark/pull/31540
+    # which helps with spurious task failures as observed in our tests. If you are running
+    # Spark versions before 3.1.1, this sets the spark.max.taskFailures to 4 to allow for
+    # more lineant configuration, else it will set them to 1 as spurious task failures are not expected
+    # for Spark 3.1.1+
+    VERSION_STRING=`$SPARK_HOME/bin/pyspark --version 2>&1|grep version\ |grep -v Scala|awk '{print $5}'`
+    echo "Detected Spark version is $VERSION_STRING"
```

Comment:
this involves some heavy initialization of Spark JVM and takes a few seconds on my machine. How about we read it from the spark core jar directly to the tune of 
```bash
unzip -q -c ~/dist/spark-3.1.1-bin-hadoop3.2/jars/spark-core_*.jar spark-version-info.properties | cut -d= -f2 | head -1
3.1.1
```

Reply:
that is way better than my prior parse/skip/ascii art solution.